plugins/message-markdown-preview/src/stuff/patcher.tsx

```diff
@@ -0,0 +1,43 @@
+import { findByProps } from "@vendetta/metro";
+import { after } from "@vendetta/patcher";
+import { findInReactTree } from "@vendetta/utils";
+import PreviewButton from "../components/PreviewButton";
+
+const { ChatInput } = findByProps("ChatInput");
+
+export let patches = [];
+
+export default () => {
+  patches.push(
+    after("render", ChatInput.prototype, (_, ret) => {
+      const props = findInReactTree(
+        ret,
+        (x) => typeof x?.placeholder === "string"
+      );
+      if (!props?.onChangeText) return;
+      const children = findInReactTree(
+        ret,
+        (x) =>
+          x?.type?.displayName === "View" && Array.isArray(x?.props?.children)
+      )?.props?.children;
+      if (!children) return;
+
+      let thing: { runner: (txt: string) => void } = {
+        runner: () => undefined,
+      };
+      if (props.onChangeText)
+        props.onChangeText = (txt: string) => thing.runner(txt);
+      else
+        patches.push(
+          after("onChangeText", props, ([txt]: [string]) => thing.runner(txt))
+        );
```

Comment:
This got me confused. Did you mean `if (!props.onChangeText)`? Patching a non-existing function here makes no sense.

Also, since this patch is ran every render, this could potentially cause a memory leak. If it is called on every text change, then it is probably safe to include `true` as the 4th argument, so that the patch is only executed once.

Reply:
true, didn't realize that, thanks

Reply:
https://github.com/Gabe616/VendettaPlugins/commit/f7ab6fb5d73e5ba25f5fa5ff719ad5d7e1d64ed7
will ask for an upsert when i join the server back